src/sentry/api/serializers/models/integration_feature.py

```diff
@@ -1,12 +1,19 @@
 from sentry.api.serializers import Serializer, register
 from sentry.models import IntegrationFeature
+from sentry.models.user import User
 
 
 @register(IntegrationFeature)
 class IntegrationFeatureSerializer(Serializer):
-    def serialize(self, obj, attrs, user):
-        return {
-            "description": obj.description.strip(),
+    def serialize(self, obj: IntegrationFeature, attrs, user: User, has_target: bool = True):
+        data = {
+            "featureId": obj.feature,
             # feature gating work done in getsentry expects the format 'featureGate'
             "featureGate": obj.feature_str(),
         }
+        if has_target:
+            # These properties require a target on the IntegrationFeature.
+            # If no target is provided, the serialized IntegrationFeature payload will be generic,
+            # and not only applicable to one target.
+            data.update({"description": obj.description.strip()})
```

Comment:
`description()` is actually making a DB call here when `obj.user_description` is null. I think it should be fine because there are so few `IntegrationFeature` rows.
But since this is used on the index endpoint and we're getting _every_ row it might be a good idea to get the names in `get_attrs()`.

Reply:
It might look something like: 
```py
def get_attrs(self, item_list: List[Any], user: Any, **kwargs: Any) -> MutableMapping[Any, Any]:
    sentry_app_ids = set()
    doc_integration_ids = set()
    for item in item_list:
        if item.target_type == IntegrationTypes.SENTRY_APP.value:
            sentry_app_ids.add(item.target_id)
        if item.target_type == IntegrationTypes.DOC_INTEGRATION.value:
            doc_integration_ids.add(item.target_id)

    sentry_app_names = SentryApp.objects.get(id__in=sentry_app_ids)
    doc_integration_names = DocIntegration.objects.get(id__in=self.doc_integration_ids)

    names_by_id_by_type = defaultdict(dict)
    for integration in sentry_app_names:
        names_by_id_by_type[IntegrationTypes.SENTRY_APP.value][
            integration.id
        ] = integration.name
    for integration in doc_integration_names:
        names_by_id_by_type[IntegrationTypes.DOC_INTEGRATION.value][
            integration.id
        ] = integration.name

    return {
        item: {"description": names_by_id_by_type[item.target_type][item.target_id]}
        for item in item_list
    }
```

Reply:
Offloaded the logic to the manager to do the bulk queries.